File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    const stack = [];
    let count = 0;

    moves.forEach((move) => {
        for (let i = 0; i < board[0].length; i++) {
            if (board[i][move - 1] === 0) continue;
            
            if (stack[stack.length - 1] === board[i][move - 1]) {
                stack.pop();
                count += 2;
            } else {
                stack.push(board[i][move - 1]);
            }
            board[i][move - 1] = 0;
            break;
        }
    })
    return count;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    newBoard = []
    for i in range(len(board)):
        newBoard.append([]) # [[], [], [], [], []]
    
    for row in range(len(board) - 1, -1, -1): # 맨 아랫줄부터 4, 3, 2, 1, 0
        for col in range(len(board)): # 0 1 2 3 4
            item = board[row][col]
            if item != 0:
                newBoard[col].append(item)
    
    count = 0
    stack = []
    for move in moves:
        if newBoard[move - 1]:
            stack.append(newBoard[move - 1].pop())
    
    count = 0
    result = []
    for item in stack:
        if result and result[-1] == item:
            result.pop()
            count += 2
        else:
            result.append(item)
    
    return count